source/grid/grid_generator_pipe_junction.cc

```diff
@@ -0,0 +1,458 @@
+    // must exceed to avoid being considered "too short". We relate this length
+    // to the longest pipe segment.
+    const double tolerance_length =
+      tolerance *
+      *std::max_element(skeleton_length.begin(), skeleton_length.end());
+
+    for (unsigned int p = 0; p < n_pipes; ++p)
+      Assert(skeleton_length[p] > tolerance_length,
+             ExcMessage("Invalid input: bifurcation matches opening."));
+
+    const auto skeleton_unit = [&]() {
+      constexpr unsigned int      n_pipes = 3;
+      std::array<vector, n_pipes> skeleton_unit;
+      for (unsigned int p = 0; p < n_pipes; ++p)
+        skeleton_unit[p] = skeleton[p] / skeleton_length[p];
+      return skeleton_unit;
+    }();
+
+    // To determine the orientation of the pipe segments to each other, we will
+    // construct a plane: starting from the bifurcation point, we will move by
+    // the magnitude one in each of the skeleton directions and span a plane
+    // with the three points we reached.
+    //
+    // The normal vector of this particular plane then describes the edge at
+    // which all pipe segments meet. If we would interpret the bifurcation as a
+    // ball joint, the normal vector would correspond to the polar axis of the
+    // ball.
+    const auto normal = [&]() {
+      const auto normal = cross_product_3d(skeleton_unit[1] - skeleton_unit[0],
+                                           skeleton_unit[2] - skeleton_unit[0]);
+      Assert(normal.norm() > tolerance_length,
+             ExcMessage("Invalid input: all three openings "
+                        "are located on one line."));
+
+      return normal / normal.norm();
+    }();
+
+    // Projections of all skeleton vectors perpendicular to the normal vector,
+    // or in other words, onto the plane described above.
+    const auto skeleton_plane = [&]() {
+      constexpr unsigned int      n_pipes = 3;
+      std::array<vector, n_pipes> skeleton_plane;
+      for (unsigned int p = 0; p < n_pipes; ++p)
+        {
+          skeleton_plane[p] = skeleton[p] - (skeleton[p] * normal) * normal;
+          Assert(std::abs(skeleton_plane[p] * normal) <
+                   tolerance * skeleton_plane[p].norm(),
+                 ExcInternalError());
+          Assert(skeleton_plane[p].norm() > tolerance_length,
+                 ExcMessage("Invalid input."));
+        }
+      return skeleton_plane;
+    }();
+
+    // Create a hyperball domain in 2D that will act as the reference cross
+    // section for each pipe segment.
+    const auto tria_base = []() {
+      constexpr unsigned int               dim      = 3;
+      constexpr unsigned int               spacedim = 3;
+      Triangulation<dim - 1, spacedim - 1> tria_base;
+      GridGenerator::hyper_ball_balanced(tria_base,
+                                         /*center=*/Point<spacedim - 1>(),
+                                         /*radius=*/1.);
+      return tria_base;
+    }();
```

Comment:
Same here, avoid re-declaring variables by removing the lambda function
```suggestion
    Triangulation<dim - 1, spacedim - 1> tria_base;
    GridGenerator::hyper_ball_balanced(tria_base,
                                       /*center=*/Point<spacedim - 1>(),
                                       /*radius=*/1.);
```